ldap-authenticator/src/main/java/org/xwiki/contrib/ldap/XWikiLDAPConfig.java

```diff
@@ -325,6 +331,11 @@ public String getLDAPParam(String name, String cfgName, String def)
             param = this.cfgConfigurationSource.getProperty(cfgName);
         }
 
+        // Look in the default configuration that might be provided by the enabled authenticator
+        if (param == null || "".equals(param)) {
```

Comment:
Regarding nr 2, we can talk about 2 "defaults": `def` which is passed in `getLDAPParam(String name, String cfgName, String def)` and the value from the `defaultConfiguration`. The `defaultConfiguration` holds the default of my custom authenticator. I want a fallback to the `def` only if my authenticator is not providing any default in the map. 
E.g.: The `def` of the `ldap_UID_attr` is `cn` and my authenticator provides `sAMAccount`, which obviously I want to be considered first. 
Also, seeing the `def` having values like `0` for the `ldap_port` or even `null` for others, makes me think that these are safety defaults to prevent NPE.